ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -1160,6 +1160,30 @@ public void shouldDistributeTerminateQuery() {
     assertThat(result.getStatementText(), is(terminateSql));
   }
 
+  @Test
+  public void shouldDistributeTerminateAllQueries() {
+    // Given:
+    createQuery(
+        "CREATE STREAM test_explain AS SELECT * FROM test_stream;",
+        emptyMap());
+
+    final String terminateSql = "TERMINATE ALL;";
+
+    // When:
+    final CommandStatusEntity result = makeSingleRequest(terminateSql, CommandStatusEntity.class);
+
+    // Then:
+    verify(commandStore)
+        .enqueueCommand(
```

Comment:
this got me thinking - if we don't want to need to support this for backwards compatibility reasons, what about distributing it as N normal terminate queries? I'm worried that this might also not play well with #3945 - what would be the key for a `TERMINATE ALL` statement?

Reply:
That's valid. I can take a look.

Reply:
I'm going to do this in a separate PR as its more convoluted.

Reply:
https://github.com/confluentinc/ksql/pull/3971